middle_end/flambda2/simplify/simplify_apply_expr.ml

```diff
@@ -417,15 +417,15 @@ let simplify_direct_partial_application ~simplify_expr dacc apply
   in
   let applied_unarized_args, _ =
     Misc.Stdlib.List.map2_prefix
-      (fun arg kind -> arg, kind)
+      (fun arg kind -> arg, K.With_subkind.kind kind)
```

Comment:
I think you should revert this part of the patch, and instead update `mk_value_slot` below to use the subkind information to set `is_always_immediate`.